data_types/src/database_name.rs

```diff
@@ -0,0 +1,147 @@
+use snafu::Snafu;
+use std::{borrow::Cow, ops::Range};
+
+/// Length constraints for a database name.
+///
+/// A `Range` is half open covering [1, 64)
+const LENGTH_CONSTRAINT: Range<usize> = 1..64;
+
+/// Database name validation errors.
+#[derive(Debug, Snafu)]
+pub enum DatabaseNameError {
+    #[snafu(display(
+        "Database name {} length must be between {} and {} characters",
+        name,
+        LENGTH_CONSTRAINT.start,
+        LENGTH_CONSTRAINT.end
```

Comment:
```suggestion
        LENGTH_CONSTRAINT.start(),
        LENGTH_CONSTRAINT.end()
```

(`RangeInclusive` doesn't expose these as fields).